source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -1187,6 +1297,25 @@ void Filter::mergePerRouteConfig() {
   }
 }
 
+void DeferredDeletableStream::closeStreamOnTimer(Filter* filter) {
+  // Close the stream.
+  if (stream_) {
+    ENVOY_LOG(debug, "Closing the stream");
+    stream_->close();
+    stream_.reset();
+  } else {
+    ENVOY_LOG(debug, "Stream already closed");
+  }
+
+  // Erase this entry from the map.
+  parent.erase(filter);
+}
+
+void DeferredDeletableStream::deferredClose(Envoy::Event::Dispatcher& dispatcher, Filter* filter) {
+  derferred_close_timer = dispatcher.createTimer([this, filter] { closeStreamOnTimer(filter); });
```

Comment:
You're going to need a bunch of comments and explanation of why we're using timers to manage stream closure here. You have some of this in your design docs I think, but all these stream semantics and deferred closes are taking an already super complicated lifecycle and making it even more complex, so we owe future readers this.